bee-api/bee-rest-api/src/endpoints/routes/api/v2/add_peer.rs

```diff
@@ -1,87 +1,55 @@
-// Copyright 2020-2021 IOTA Stiftung
+// Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use std::net::IpAddr;
-
-use bee_gossip::{Command::AddPeer, Multiaddr, NetworkCommandSender, PeerId, PeerRelation, Protocol};
-use bee_protocol::workers::PeerManager;
-use bee_runtime::resource::ResourceHandle;
-use serde_json::Value as JsonValue;
-use warp::{filters::BoxedFilter, http::StatusCode, reject, Filter, Rejection, Reply};
+use axum::{
+    extract::{Extension, Json},
+    response::IntoResponse,
+    routing::post,
+    Router,
+};
+use bee_gossip::{Command::AddPeer, Multiaddr, PeerId, PeerRelation, Protocol};
+use log::error;
+use serde_json::Value;
 
 use crate::{
-    endpoints::{
-        config::ROUTE_ADD_PEER,
-        filters::{with_network_command_sender, with_peer_manager},
-        permission::has_permission,
-        rejection::CustomRejection,
-    },
+    endpoints::{error::ApiError, storage::StorageBackend, ApiArgsFullNode},
     types::{
         dtos::{PeerDto, RelationDto},
         responses::AddPeerResponse,
     },
 };
 
-fn path() -> impl Filter<Extract = (), Error = warp::Rejection> + Clone {
-    super::path().and(warp::path("peers")).and(warp::path::end())
-}
-
-pub(crate) fn filter(
-    public_routes: Box<[String]>,
-    allowed_ips: Box<[IpAddr]>,
-    peer_manager: ResourceHandle<PeerManager>,
-    network_command_sender: ResourceHandle<NetworkCommandSender>,
-) -> BoxedFilter<(impl Reply,)> {
-    self::path()
-        .and(warp::post())
-        .and(has_permission(ROUTE_ADD_PEER, public_routes, allowed_ips))
-        .and(warp::body::json())
-        .and(with_peer_manager(peer_manager))
-        .and(with_network_command_sender(network_command_sender))
-        .and_then(
-            |value, peer_manager, network_controller| async move { add_peer(value, peer_manager, network_controller) },
-        )
-        .boxed()
+pub(crate) fn filter<B: StorageBackend>() -> Router {
+    Router::new().route("/peers", post(add_peer::<B>))
 }
 
-pub(crate) fn add_peer(
-    value: JsonValue,
-    peer_manager: ResourceHandle<PeerManager>,
-    network_controller: ResourceHandle<NetworkCommandSender>,
-) -> Result<impl Reply, Rejection> {
+pub(crate) async fn add_peer<B: StorageBackend>(
+    Json(value): Json<Value>,
+    Extension(args): Extension<ApiArgsFullNode<B>>,
+) -> Result<impl IntoResponse, ApiError> {
     let multi_address_v = &value["multiAddress"];
```

Comment:
I personally think `multi_addr_val` or `multi_addr_value` are more readable abbreviations, but I would even consider not abbreviating at all: `multi_address_value`. Your choice though :smile: 
(same for alias below)

Reply:
@Alex6323 should we maybe even add the suffix `_json` instead? would be even more clear I guess